docs: Docs nav structure reorg follow-up (#1407)
meltano/sdk: docs/development/release_process.md

## PyPI releases

> Releases are published to PyPi by a GitHub Actions workflow, which in turn is triggered by a GitHub [Release](https://github.com/meltano/sdk/releases) being published.

Reviewer: ```suggestion
Releases are published to PyPI by a GitHub Actions workflow, triggered when a GitHub [Release](https://github.com/meltano/sdk/releases) is published.
```
